File: content/04/code/java/04-empty-scenarios-added/shouty/src/test/java/shouty/StepDefinitions.java
package shouty;

import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

import static java.util.Arrays.asList;
import static org.junit.Assert.assertEquals;

public class StepDefinitions {

    private Person sean;
    private Person lucy;
    private String messageFromSean;

    @Given("Lucy is {int} metres from Sean")
    public void lucy_is_located_m_from_Sean(int distance) throws Throwable {
        Network network = new Network();
        sean = new Person(network);
        lucy = new Person(network);
        lucy.moveTo(distance);
    }

    @When("Sean shouts {string}")
    public void sean_shouts(String message) throws Throwable {
        sean.shout(message);
        messageFromSean = message;
    }

    @Then("Lucy should hear Sean's message")
    public void lucy_hears_Sean_s_message() throws Throwable {
        assertEquals(asList(messageFromSean), lucy.getMessagesHeard());
    }
}

File: content/bdd-with-cucumber/04/code/dotnet/05--step05-snippets-added/Shouty/Shouty.Specs/StepDefinitions/StepDefinitions.cs
using System;
using TechTalk.SpecFlow;
using Xunit;

namespace Shouty.Specs.StepDefinitions
{
    [Binding]
    public class StepDefinitions
    {
        private Person lucy;
        private Person sean;
        private string messageFromSean;

        [Given("Lucy is {int} metres from Sean")]
        public void GivenLucyIsMetresFromSean(int distance)
        {
            var network = new Network();
            sean = new Person(network);
            lucy = new Person(network);
            lucy.MoveTo(distance);
        }

        [Given("a person named Lucy")]
        public void GivenAPersonNamedLucy()
        {
            throw new PendingStepException();
        }

        [Given("a person named Sean")]
        public void GivenAPersonNamedSean()
        {
            throw new PendingStepException();
        }

        [When("Sean shouts {string}")]
        public void WhenSeanShouts(string message)
        {
            sean.Shout(message);
            messageFromSean = message;
        }

        [Then("Lucy should hear Sean's message")]
        public void ThenLucyShouldHearSeansMessage()
        {
            Assert.Contains(messageFromSean, lucy.GetMessagesHeard());
        }
    }
}
